diff --git a/api/oc_ri.c b/api/oc_ri.c
index 3294608..28fee2f 100644
--- a/api/oc_ri.c
+++ b/api/oc_ri.c
@@ -899,7 +899,7 @@ oc_ri_invoke_client_cb(void *response, oc_endpoint_t *endpoint)
           break;
         }
       }
-      coap_get_header_observe(pkt, &client_response.observe_option);
+      coap_get_header_observe(pkt, (uint32_t*)&client_response.observe_option);
 
       bool separate = false;
       /*
diff --git a/api/oc_uuid.c b/api/oc_uuid.c
index 2ceaf7a..8e6fde5 100644
--- a/api/oc_uuid.c
+++ b/api/oc_uuid.c
@@ -30,7 +30,7 @@ oc_str_to_uuid(const char *str, oc_uuid_t *uuid)
   for (i = 0; i < strlen(str); i++) {
     if (str[i] == '-')
       continue;
-    else if (isalpha(str[i])) {
+    else if (isalpha((int)str[i])) {
       switch (str[i]) {
       case 65:
       case 97:
diff --git a/deps/tinycbor b/deps/tinycbor
--- a/deps/tinycbor
+++ b/deps/tinycbor
@@ -1 +1 @@
-Subproject commit 31c7f81d45d115d2007b1c881cbbd3a19618465c
+Subproject commit 31c7f81d45d115d2007b1c881cbbd3a19618465c-dirty
diff --git a/util/oc_list.c b/util/oc_list.c
index 429f411..1eee62f 100644
--- a/util/oc_list.c
+++ b/util/oc_list.c
@@ -34,7 +34,9 @@
 
 #include "oc_list.h"
 
+#ifndef NULL
 #define NULL 0
+#endif
 
 struct list
 {
