diff --git a/src/compilersupport_p.h b/src/compilersupport_p.h
index 3c24ef1..6cee5c7 100644
--- a/src/compilersupport_p.h
+++ b/src/compilersupport_p.h
@@ -61,7 +61,9 @@
 #  define inline    CBOR_INLINE
 #endif
 
+#ifndef STRINGIFY
 #define STRINGIFY(x)            STRINGIFY2(x)
+#endif
 #define STRINGIFY2(x)           #x
 
 #ifndef UINT32_MAX
@@ -150,8 +152,12 @@
 #endif
 
 #ifdef __GNUC__
+#ifndef likely
 #  define likely(x)     __builtin_expect(!!(x), 1)
+#endif
+#ifndef unlikely
 #  define unlikely(x)   __builtin_expect(!!(x), 0)
+#endif
 #  define unreachable() __builtin_unreachable()
 #elif defined(_MSC_VER)
 #  define likely(x)     (x)
